wandb/sdk/launch/agent/agent.py

```diff
@@ -596,13 +600,30 @@ def _check_run_finished(self, job_tracker: JobAndRunStatusTracker) -> bool:
         try:
             run = job_tracker.run
             status = run.get_status().state
-            if status in ["stopped", "failed", "finished"]:
+
+            # TODO change these statuses to an enum
+            if status in ["stopped", "failed", "finished", "preempted"]:
```

Comment:
:(

Reply:
Left there by request of Griffin, I think changing them to an enum would bloat this PR but it would make a good followup

Reply:
In core we have separate conditions for terminal and non-terminal based on the state. We hold these in separate types, we should do something similar here (in a separate PR)